.github/workflows/stale.yml

```
@@ -31,6 +31,8 @@ jobs:
             Hi everyone! This issue has been closed due to inactivity.
             If you think this issue is still relevant in the latest Solidity version and you have something to [contribute](https://docs.soliditylang.org/en/latest/contributing.html), feel free to reopen.
             However, unless the issue is a concrete proposal that can be implemented, we recommend starting a language discussion on the [forum](https://forum.soliditylang.org) instead.
+          any-of-issue-labels: stale # FIXME: remove this when debug is false
+          ascending: true # FIXME: remove this when debug is false
```

`FIXME` to me implies that it's broken in some way.
```suggestion
          any-of-issue-labels: stale # TODO: remove this when debug is false
          ascending: true # TODO: remove this when debug is false
```

Same with the one above.

---

Actually, why would we remove it? Isn't this exactly what we want?

---

Not the `any-of-issue-labels` since the action should also label the issues, which it will not do in the current setting. The `ascending` order may be desirable indeed, but once we clean up the old issues, I think we will want to keep the action running from the most recent to the old ones, no?

---

> once we clean up the old issues, I think we will want to keep the action running from the most recent to the old ones, no?

Why? It makes perfect sense to me to start with the oldest ones as they're more likely to be actually stale and need closing.

---

> Not the `any-of-issue-labels` since the action should also label the issues, which it will not do in the current setting. 

I'm not sure I understand. The action is now configured to label issues as `stale` after 180 days and we have plenty of those. Why wouldn't it label them? (and BTW, wasn't it supposed to be 11 months so that only pre-2022 issues count for now? We'll only be moving on to 6-months old in the next step).

And even assuming that for some reason it would not label them - what does it matter if we have issues manually labelled as `stale` now and we're only concerned with those at this stage?

Finally, "when debug is false" means when #13756 is merged. Which is not the next stage. This is still the same stage. I don't understand why you want to test the action with one set of settings and then actually run it with different settings. I also don't see you removing these TODOs in #13756.

---

> Why? It makes perfect sense to me to go from oldest to newest as the old ones are more likely to be actually stale and need closing.

Sure, but I guess the ascending order is useful for what we want to do now, i.e., close as much as possible of very old issues. However, once we get to the timeframe that we want (the 11 months that you mentioned), I think it would be better to focus on the recent issues, no? Because these are the ones we care about, we do not want them to get stale. If the action only focuses on old ones, the current scenario may remain the same, i.e., piling up issues due to the low throughput.


> The action is now configured to label issues as `stale` after 180 days and we have plenty of those. Why wouldn't it label them? (and BTW, wasn't it supposed to be 11 months so that only pre-2022 issues count for now? We'll only be moving on to 6 months old in the next step).

Because as the [any-of-issue-labels](https://github.com/actions/stale#any-of-issue-labels) is set to `stale` the action will only process issues that are already labeled as `stale`. As @NunoFilipeSantos already labeled the issues older than 11 months ago and as the action has the [days-before-issue-close](https://github.com/actions/stale#days-before-issue-close) set to `0`, my understanding was that it would automatically close any issue labeled as `stale` (up to the maximum operations per run). But I may be wrong. This is why I thought that the 6 months period wouldn't be relevant now, but we can change it to 11 months, just to be sure.


> what does it matter if we have issues manually labelled as `stale` now and we're only concerned with those at this stage?

Indeed it does not matter for the general case. I just thought that the idea now was only close to what was already labeled. If the action needs to work on any issue and not only close what is already labeled, I can revert that change. Actually, it is kind of how it is now in the `develop` branch because it considers the issues that are 180 days old but in descending order.


> Finally, "when debug is false" means when #13756 is merged. Which is not the next stage. This is still the first stage. 

True, that ` "when debug is false"` comment is indeed misleading. Those instructions were supposed to be removed once we finish closing all ancient issues.


> I don't understand why you want to test the action with one set of settings and then actually run it with different settings.

What do you mean? Wasn't that what you and @NunoFilipeSantos suggested in the Project Management channel? Sorry, I may have misunderstood the idea. But here are the quotes from the chat which I took as a base for the initial PR:
```
@cameel : I think we should use the same action for both.
I.e. it's a pretty generic stale issue action and until we close off old stuff, we don't want it to mark stuff as stale yet anyway.
So we can configure to do the hard closing in the initial PR and then, when the action is done with closing,
reconfigure it in another PR to be a stale bot with stale period set to 6 months.
...
@NunoFilipeSantos: Close every single issue that has been signaled as stale now. No batches.
The rest we can close 50 a day (we have 1000) after the 2 said weeks are complete.
The important bit is closing them as stale in the action not normal closing.
This will help if we need to go back in some of them.
```

>  I also don't see you removing these TODOs in #13756.

It is because turning the debug to `false` would not remove the TODOs yet, but only after we close all the old stale issues. The comments are indeed wrong.

---

> I think it would be better to focus on the recent issues, no? Because these are the ones we care about, we do not want them to get stale.

Ah, that's how you see it. I don't think it's important to get to any of the stale issues quickly - otherwise we should just decrease the stale period. Anyway, either way is fine with me, just wanted to know why you thought it matters.

> Because as the [any-of-issue-labels](https://github.com/actions/stale#any-of-issue-labels) is set to `stale` the action will only process issues that are already labeled as `stale`.

Right. I see it now. It won't label anything that's not already labeled `stale`. Yeah, that indeed needs to be removed in the next stage.

> This is why I thought that the 6 months period wouldn't be considered now by the action, but we can change it to 11 months, just to be sure.

Oh, right. So the 180 days is not used for now. Even if it does not take effect, I think it would still be better to have 11 months there just to avoid confusion and change it only when we get to the next stage.

> True, that ` "when debug is false"` comment is indeed misleading. Those instructions were supposed to be removed once we finish closing all ancient issues.

Let's fix the comment then.

> What do you mean? This PR here is just a fix, because the current merged one is running on the wrong set of issues.

I meant that according to the TODO should change settings when we change `debug` to `false`. Now I see that's not what you meant.

---

> Ah, that's how you see it. I don't think it's important to get to any of the stale issues quickly - otherwise we should just decrease the stale period. 

Indeed. Either way is fine by me as well. We can easily change that later anyway.

> Right. I see it now. It won't label anything that's not already labeled `stale`. Yeah, that indeed needs to be removed in the next stage.

Yes, this is my understanding from the [stale](https://github.com/actions/stale) action documentation, and indeed seems to be the case in some local tests that I did.


> Oh, right. So the 180 days is not used for now. Even if it does not take effect, I think it would still be better to have 11 months there just to avoid confusion and change it only when we get to the next stage.

I update it to roughly 11 months (~334 days).